### PR TITLE
Reduce blocking the calling thread when calculating default user-agent string

### LINKED ISSUE #615: -[GTLRService requestUserAgent] blocks the calling thread in GTMFetcherApplicationIdentifier()
By default, `-[GTLRService executeQuery:completionHandler:]` blocks the calling thread to invoke `GTMFetcherApplicationIdentifier()`:

https://github.com/google/google-api-objectivec-client-for-rest/blob/a91e1a0fad3740f42fa214cef9e60459ac9fd80a/Sources/Core/GTLRService.m#L290

That method is synchronized on a process-wide mutex:

https://github.com/google/gtm-session-fetcher/blob/ffb0463f5e57d250c101eca46fc9cb5a0035dc6c/Sources/Core/GTMSessionFetcher.m#L4811

This causes stalls, hangs, and watchdog terminations when `GTLRService` is invoked from the UI thread to asynchronously execute a query, since another thread can be busy reading all the data from disk / other services to calculate the User-Agent string.

Last year, I filed and fixed https://github.com/google/gtm-session-fetcher/issues/352 on the `GTMSessionFetcher` side to allow doing this work asynchronously. I enabled it by default in https://github.com/google/gtm-session-fetcher/pull/357 . We should make use of that in `GTLRService` as well.

### PR DESCRIPTION
**Pull request by bhamiltoncx**
Fixes #615.

### CODE REVIEW

File: Source/Objects/GTLRService.m [hunk 1]
```diff
@@ -338,16 +390,17 @@ - (NSString *)requestUserAgent {
...
-        || [owningBundle.bundleIdentifier isEqual:@"com.google.GTLR"]) {
-      owningBundle = [NSBundle mainBundle];
-    }
-    userAgent = GTMFetcherApplicationIdentifier(owningBundle);
+    //
+    // This blocks until the prefetch in the initializer completes.
+    dispatch_sync(DefaultUserAgentQueue(), ^{
+      GTLR_DEBUG_ASSERT(_defaultUserAgent != nil,
+                        @"_defaultUserAgent must have been fetched by now");
+      userAgent = _defaultUserAgent;
```
Comment (thomasvl) [inline Source/Objects/GTLRService.m:402]: I guess we could also skip the new ivar and just call `DefaultUserAgentForServiceClass()` again here. Did you see any explicit reason it was better to cache it?

**Comment by thomasvl** [comment]
Can you also do a sync, it looks like there's a conflict on GTLRService.m.

**Comment by bhamiltoncx** [comment]
> Can you also do a sync, it looks like there's a conflict on GTLRService.m.

Whoops, fixed.

**Comment by thomasvl** [comment]
Leaning in the directions of #623 because it seems to better put the work at first query time instead of at Service creation, so that seems a little better.

**Comment by bhamiltoncx** [comment]
Cool, I'm happier with #623 as well. Closing this PR in favor of that one.

**Pull request closed without merging**